.vscode/launch.json

```
@@ -182,6 +182,33 @@
                 "<node_internals>/**"
             ]
         },
+        {
+            "name": "Tests (DataScience, *.ds.test.ts)",
+            "type": "extensionHost",
+            "request": "launch",
+            "runtimeExecutable": "${execPath}",
+            "args": [
+                "${workspaceFolder}/src/test",
+                "--disable-extensions",
+                "--enable-proposed-api",
+                "--extensionDevelopmentPath=${workspaceFolder}",
+                "--extensionTestsPath=${workspaceFolder}/out/test"
+            ],
+            "env": {
+                "VSC_PYTHON_CI_TEST_GREP": "xxx", // Modify this to run a subset of the single workspace tests
+                "VSC_PYTHON_LOAD_EXPERIMENTS_FROM_FILE": "true",
+                "TEST_FILES_SUFFIX": "ds.test"

```

I'm thinking of using this for ongoing data science tests.
* Our ds tests run in a separate job (PR will fail faster)
* We can make this a mandatory job in CI later (which we need)
* Also will test against actual extension bits (we don't need to construct IOC).
* Faster

There might be a few rare occasions where we might want to inject a new item into the container, however I think thats possible as well, lets leave that for when its necessary.

---

That only works for using real jupyter. Not a great idea on a PR. At least not yet.

---
In reply to: [428119607](https://github.com/microsoft/vscode-python/pull/11910#discussion_r428119607) [](ancestors = 428119607)